packages/tools/client-debugger/client-debugger-view/src/components/DataObjectsView.tsx

```diff
@@ -18,10 +32,61 @@ export type DataObjectsViewProps = HasContainerId;
  * Dispatches data object rendering based on those provided view {@link DataObjectsViewProps.renderOptions}.
  */
 export function DataObjectsView(props: DataObjectsViewProps): React.ReactElement {
+	const { containerId } = props;
+
+	const messageRelay = useMessageRelay();
+
+	const [rootDataHandles, setRootDataHandles] = React.useState<
+		Record<string, RootHandleNode> | undefined
+	>();
+
+	React.useEffect(() => {
+		const inboundMessageHandlers: InboundHandlers = {
+			["ROOT_DATA_VISUALIZATIONS"]: (untypedMessage) => {
+				const message: RootDataVisualizationsMessage =
+					untypedMessage as RootDataVisualizationsMessage;
+
+				if (message.data.containerId === containerId) {
+					setRootDataHandles(message.data.visualizations);
+
+					return true;
+				} else {
+					return false;
+				}
+			},
+		};
+
+		function messageHandler(message: Partial<IDebuggerMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers, {
+				context: loggingContext,
+			});
+		}
+
+		messageRelay.on("message", messageHandler);
+
+		messageRelay.postMessage({
+			type: "GET_ROOT_DATA_VISUALIZATIONS",
+			data: {
+				containerId,
+			},
+		});
+
+		return (): void => {
+			messageRelay.off("message", messageHandler);
+		};
+	}, [containerId, rootDataHandles, messageRelay]);
+
+	if (rootDataHandles === undefined) {
+		return <Waiting label="Waiting for container DDS data. DATAOBJECTSVIEW" />;
```

Comment:
We should remove the "DATAOBJECTSVIEW" part before merging.

We may also want to extract the label text out as a constant somewhere so our different data view components can re-use it.

Reply:
Created additional component `waitingLabels` which contains the constant string warning labels.